rpc/client.go

```diff
@@ -265,18 +265,19 @@ func (c *Client) heartbeat() error {
 				remoteTimeNow := response.ServerTime + (receiveTime-sendTime)/2
 				c.offset.Offset = remoteTimeNow - receiveTime
 			}
+			offset := c.offset
 			c.mu.Unlock()
-			c.remoteClocks.UpdateOffset(c.addr.String(), c.offset)
+			c.remoteClocks.UpdateOffset(c.addr.String(), offset)
 		}
 		return call.Error
 	case <-time.After(heartbeatInterval * 2):
-		// Allowed twice gossip interval.
+		// Allowed twice heartbeat interval.
 		c.mu.Lock()
 		c.healthy = false
 		c.offset = proto.InfiniteOffset
-		c.offset.MeasuredAt = c.clock.PhysicalNow()
+		offset := c.offset
 		c.mu.Unlock()
-		c.remoteClocks.UpdateOffset(c.addr.String(), c.offset)
+		c.remoteClocks.UpdateOffset(c.addr.String(), offset)
```

Comment:
UpdateOffset will always be a no-op here because we set c.offset to InfiniteOffset. Can we just remove everything but the logging from this `case`?